holmes/plugins/toolsets/logging_api.py

```diff
@@ -0,0 +1,205 @@
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+import logging
+from typing import Optional
+
+from pydantic import BaseModel
+from datetime import timezone
+from holmes.core.tools import (
+    StructuredToolResult,
+    Tool,
+    ToolParameter,
+    Toolset,
+)
+from holmes.plugins.toolsets.utils import get_param_or_raise
+
+# Default values for log fetching
+DEFAULT_LOG_LIMIT = 2000
+DEFAULT_TIME_SPAN_SECONDS = 3600
+
+
+class LoggingLabelsConfig(BaseModel):
+    """Common label mapping configuration for all logging backends"""
+
+    pod: Optional[str] = None
+    namespace: Optional[str] = None
+
+
+class LoggingConfig(BaseModel):
+    """Base configuration for all logging backends"""
+
+    labels: Optional[LoggingLabelsConfig] = None
+
+
+class FetchLogsParams(BaseModel):
+    namespace: str
+    pod_name: str
+    start_time: Optional[str] = None
+    end_time: Optional[str] = None
+    filter_pattern: Optional[str] = None
+    limit: Optional[int] = None
+
```

Comment:
I think that since pod_name is required it means that this is specificly for fetching pod logs. Can we make it specific? Do you think it is good idea? are you hacing future plan and this is why the name is general?
```suggestion
class FetchPodLogsParams(BaseModel):
    namespace: str
    pod_name: str
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    filter_pattern: Optional[str] = None
    limit: Optional[int] = None

```

Reply:
I probably wasn't clear enough. What i meant is that from my pov we need to change the tool to be called FetchPodLogs, and align all names to it. 
The main reason is we agreed the the api will be very specific, we want to start with the most simple and naive implementation and improve from there. LogsToolset is a good name but the tools names should specific to the cases. e.g. FetchPodLogs, FetchDeploymentLogs etc. Otherwise how the LLM will know to select the relevant tools?

The the result will be that we have LogsToolset which has all relevant api (tools) we expect
LogsToolset
- FetchPodLogs
- FetchDeploymentLogs
- FetchStatefulesetLogs 
- etc

And all Specific implementation logs toolset will need to implement those capabilities.

Reply:
I think the more likely evolution for this tool is to still be called `fetch_logs` and have the key arguments be `resource_type` and `resource_name`.

Reply:
And in that case it we wouldn't name it something more specific like FetchK8SLogs or FetchResourceLogs? I think that the whole idea of this task was to make thing very simple and straight forward. FetchLogs is to generic.